pkg/transport/generic_transport_controller.go

```diff
@@ -605,7 +609,7 @@ func setLabel(object metav1.Object, key string, value any) {
 }
 
 // cleanObject is a function to clean object before adding it to a wrapped object. these fields shouldn't be propagated to WEC.
-func cleanObject(object *unstructured.Unstructured) *unstructured.Unstructured {
+func (c *genericTransportController) cleanObject(object *unstructured.Unstructured) *unstructured.Unstructured {
```

Comment:
Changing this from a free-standing func to a method on a controller is an unnecessary entanglement; it thwarted the approach I was taking in #1816 of cleaning the test objects as they were generated (which happened before the controller is created). I see that there is no meaningful difference among instances of ObjectDenaturingMap, no need to hold one in a field of the controller rather than hold one in a global variable; these things have no state that varies after construction.